File: android/src/main/java/com/rnopentok/RNDefaultCapturerFactory.java
package com.rnopentok;

import android.view.View;

import com.facebook.react.uimanager.util.ReactFindViewUtil;
import com.opentok.android.BaseVideoCapturer;

public class RNDefaultCapturerFactory implements RNCapturerFactory {
    @Override
    public BaseVideoCapturer fetchCapturer(RNOpentokPublisherConfiguration configuration) throws RNPublisherUncreatable {
        if (configuration.isScreenCapture()) {
            View captureView = ReactFindViewUtil.findView(configuration.getRootView(), "RN_OPENTOK_SCREEN_CAPTURE_VIEW");
            if (captureView == null) {
                throw new RNPublisherUncreatable();
            }
            return  new RNOpenTokScreenSharingCapturer(captureView, configuration.getCaptureSettings());
        } else {
            return null; //Defaults to video capturer if null
        }
    }
}

File: ios/RNDefaultCapturerFactory.m
#import "RNDefaultCapturerFactory.h"
#import "RNOpentokPublisherConfiguration.h"
#import "RNOpenTokScreenSharingCapturer.h"
#import "RCTUtils.h"
#import "RNDefaultPublisherFactory.h"
#import "RCTUtils.h"
#import "RNOpenTokScreenSharingCapturer.h"
#import "RNOpentokPublisherConfiguration.h"
#import "RCTUIManager.h"
#import "RNCapturerFactory.h"
#import "RNDefaultCapturerFactory.h"

#if __has_include(<React/UIView+React.h>)
#import "React/UIView+React.h"
#elif __has_include("UIView+React.h")
#import <React/UIView+React.h>
#else
#import "React/UIView+React.h"
#endif

@protocol OTVideoCapture;

@implementation RNDefaultCapturerFactory {

}
- (id<OTVideoCapture>) fetchCapturer:(nonnull RNOpentokPublisherConfiguration *)configuration {
    if(configuration.screenCapture){
        UIView* rootView = RCTPresentedViewController().view;
        UIView* screenCaptureView = [configuration.uiManager viewForNativeID:@"RN_OPENTOK_SCREEN_CAPTURE_VIEW"
                                                                 withRootTag:configuration.reactTag];
        if (screenCaptureView) {
            RNOpenTokScreenSharingCapturer* capture = [[RNOpenTokScreenSharingCapturer alloc]
                    initWithView:screenCaptureView
                    withSettings: configuration.screenCaptureSettings];
            return capture;
        } else {
            [[NSNotificationCenter defaultCenter]
                    postNotificationName:@"errorNoScreenCaptureView"
                                  object:nil];
        }
    }
    return nil;//Defaults to video capturer if null
}

@end